internal/pkg/gateway/endorsement.go

```diff
@@ -8,14 +8,17 @@ package gateway
 
 import (
 	"bytes"
+	b64 "encoding/base64"
+	"errors"
 	"sync"
 
 	"github.com/hyperledger/fabric-protos-go/peer"
 )
 
 type layout struct {
-	required     map[string]int // group -> quantity
-	endorsements []*peer.ProposalResponse
+	required        map[string]int // group -> quantity
+	endorsements    []*peer.Endorsement
+	uniqueEndorsers map[string]struct{}
```

Comment:
I suggest that we de-duplicate certificates afterwards and not during the collection, as it will make everything easier to read.

Reply:
OK